### PR TITLE
Cufflinks RNA filtering

### PR DESCRIPTION
**Pull request by iskandr**
Helper functions for loading Cufflinks `fpkm_tracking` files, filtering collections by expression, and misc. fixes.

* API change: changed `gene_names()`, `gene_ids()`, `genes()`, `transcripts()` to memoized properties, added `transcript_names`, `transcript_ids`. 

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/varcode/73)

### CODE REVIEW

File: test/test_load_cufflinks_fpkm.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+"""
+test_cufflinks : Test that we can correctly load Cufflinks tracking files which
+contain the estimated expression levels of genes and isoforms (computed from
+RNA-Seq reads).
+"""
+
+<<<<<<< HEAD
```
Comment (timodonnell) [inline test/test_load_cufflinks_fpkm.py]: merge gibbberish

File: test/test_load_cufflinks_fpkm.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+from varcode.rna import load_cufflinks_tracking_file
+=======
+from varcode.cufflinks import load_cufflinks_tracking_file
+>>>>>>> de7b6d57bfee1072b528a2e15f94443500485a52
+from nose.tools import eq_
+
+def test_load_cufflinks_genes():
+    genes_df = load_cufflinks_tracking_file("data/genes.fpkm_tracking")
```
Comment (timodonnell) [inline test/test_load_cufflinks_fpkm.py]: This should use the `data_path` helper defined in `test/__init__.py` so the tests work regardless of current working directory. See `test/test_read_evidence.py` for an example.

File: varcode/collection.py [hunk 3]
```diff
@@ -14,6 +14,7 @@
 
 from __future__ import print_function, division, absolute_import
 import os.path
+from collections import defaultdict
 
 class Collection(object):
     """
```
Comment (timodonnell) [inline varcode/collection.py]: A docstring for this class would be helpful

File: varcode/rna.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        if fpkm >= fpkm_threshold
+    }
+
+def choose_principal_transcripts(
+        variant_collection,
+        gene_fpkm_df,
+        gene_expression_threshold,
+        transcript_fpkm_df,
+        transcript_expression_threshold=0.0):
+
```
Comment (timodonnell) [inline varcode/rna.py]: A one sentence docstring saying what this does would be helpful

File: varcode/rna/principal_transcripts.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
+from .combine_genes import combine_gene_expression_levels
+
+def choose_principal_transcripts(
```
Comment (timodonnell) [inline varcode/rna/principal_transcripts.py]: is this mid-refactor? there's another `choose_principal_transcripts` defined above in `rna.py`. or are these different? (sorry if missing something obvious)

**Comment by timodonnell** [comment]
see comments, but otherwise lgtm

**Comment by iskandr** [comment]
Sorry, there was some obviously broken/repeated code because I hadn't pushed my updates.

**Pull request merged**